auth/realm/base/src/main/java/org/wildfly/security/auth/realm/LegacyPropertiesSecurityRealm.java

```diff
@@ -407,6 +407,7 @@ public static Builder builder() {
          * @return this {@link Builder}
          */
         public Builder setProviders(Supplier<Provider[]> providers) {
+            assertNotNull(providers);
```

Comment:
Instead of `assertNotNull`, we could use `Assert.checkNotNullParam("providers", providers)`

Reply:
Sure, actually I first used checkNotNullParam but later realized that other two setters in Builder used asserNotNull and updated to this. Is checkNotNullParam preferred? If so, I can update the existing ones also to this

Reply:
Ah, I see. Yes, we typically use `checkNotNullParam` for this type of thing so we can also include the parameter name as a string. You could update the existing ones too.

Reply:
Sounds good. I will update them.

Reply:
Updated to use checkNotNullParam